Add description about tidb_auth_token authentication (#11119)
pingcap/docs: tidb-configuration-file.md

- Default value: "", which allows TLSv1.1 or higher.
- Optional values: `"TLSv1.0"`, `"TLSv1.1"`, `"TLSv1.2"` and `"TLSv1.3"`

### `auth-token-jwks` <span class="version-mark">New in v6.4.0</span>

> **Warning:**
>

> > The `tidb_auth_token` authentication method is used only for the internal operation of TiDB Cloud. **DO NOT** set the variable to this value.

Reviewer: ```suggestion
> The `tidb_auth_token` authentication method is used only for the internal operation of TiDB Cloud. **DO NOT** change the value of this configuration.
```

> - Set the local file path of the JSON Web Key Sets (JWKS) of the `tidb_auth_token` authentication method.

Reviewer 2: ```suggestion
- Set the local file path of the JSON Web Key Sets (JWKS) for the `tidb_auth_token` authentication method.
```

- Default value: `""`

### `auth-token-refresh-interval` <span class="version-mark">New in v6.4.0</span>

> **Warning:**
>

> > The `tidb_auth_token` authentication method is used only for the internal operation of TiDB Cloud. **DO NOT** set the variable to this value.

Reviewer: ```suggestion
> The `tidb_auth_token` authentication method is used only for the internal operation of TiDB Cloud. **DO NOT** change the value of this configuration.
```

> - Set the JSON Web Key Sets refresh interval of the `tidb_auth_token` authentication method.

Reviewer 2: ```suggestion
- Set the JWKS refresh interval for the `tidb_auth_token` authentication method.
```
